project/JavalibIRCleaner.scala

```diff
@@ -657,7 +657,11 @@ object JavalibIRCleaner {
       ClassName("scala.Tuple" + n) -> ClassName("java.util.internal.Tuple" + n)
     }
 
-    val allPairs = functionTypePairs ++ refPairs ++ tuplePairs
+    val otherPairs = List(
+      ClassName("scala.MatchError") -> ClassName("java.lang.AssertionError"),
```

Comment:
Consider putting the commit comment as code comment here.

> `AssertionError` conveniently features a constructor taking an
`Object`. Since any `MatchError` in the javalib would be a bug, it
is fine to rewrite them to `AssertionError`s.

Reply:
Good point. Done.